### PR TITLE
fix: metadata structure correction

### PR DESCRIPTION
**Pull request by sanpj2292**
## What are the changes introduced in this PR?

- Need to refer getErrorRespEvents from integrations-lib having defined types
- When destEvents (in routerTransform --> processRouterDest) is not sent as [ {}, {}, … ], instead of setting metadata as null, we will try to put it as `[]`

## What is the related Linear task?

Resolves INT-1579

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3119/test-report.html

File: test/integrations/component.test.ts [hunk 1]
```diff
@@ -43,6 +43,7 @@ command
   .option('-id, --id <string>', 'Enter unique "Id" of the test case you want to run')
   .parse();
 
+jest.setTimeout(30000000);
```
Comment (koladilip) [inline test/integrations/component.test.ts:46]: is it required?

**Review by koladilip — APPROVED** [review]

File: test/integrations/component.test.ts (hunk 1 above)
Comment (sanpj2292) [reply test/integrations/component.test.ts:46]: My bad! This was done for debugging purposes

Comment (sanpj2292) [reply test/integrations/component.test.ts:46]: Will revert

Comment (sanpj2292) [reply test/integrations/component.test.ts]: Removed this. Resolving the comment

**Review by koladilip — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**